internal/consensus/byzantine_test.go

```diff
@@ -275,12 +280,11 @@ func TestByzantinePrevoteEquivocation(t *testing.T) {
 	require.NoError(t, err)
 
 	for idx, ev := range evidenceFromEachValidator {
-		if assert.NotNil(t, ev, idx) {
-			ev, ok := ev.(*types.DuplicateVoteEvidence)
-			assert.True(t, ok)
-			assert.Equal(t, pubkey.Address(), ev.VoteA.ValidatorAddress)
-			assert.Equal(t, prevoteHeight, ev.Height())
-		}
+		require.NotNil(t, ev, idx)
```

Comment:
This will skip the rest of the validators. If you don't like the assert-check, maybe a subtest?

Reply:
That's fair. Thought I think if one fails, they're all going to fail (I think?) So I guess I'd err on the side of reporting a single error in these cases, rather than reporting many errors for non-orthogonal failures.